src/ppx/xr_component.cpp

```diff
@@ -434,11 +549,26 @@ void XrComponent::EndFrame(const std::vector<grfx::SwapchainPtr>& swapchains, ui
     XrEnvironmentBlendMode                     blendMode = mBlendModes[0];
     std::vector<XrCompositionLayerBaseHeader*> layers;
     XrCompositionLayerProjection               compositionLayerProjection = {XR_TYPE_COMPOSITION_LAYER_PROJECTION};
-    compositionLayerProjection.layerFlags                                 = 0;
+    compositionLayerProjection.layerFlags                                 = XR_COMPOSITION_LAYER_BLEND_TEXTURE_SOURCE_ALPHA_BIT;
     compositionLayerProjection.space                                      = mRefSpace;
     compositionLayerProjection.viewCount                                  = static_cast<uint32_t>(GetViewCount());
     compositionLayerProjection.views                                      = compositionLayerProjectionViews.data();
     if (mShouldRender) {
+        if (mPassthroughSupported && mPassthroughEnabled) {
+#if defined(PPX_XR_QUEST)
+            if (mPassthroughLayer != XR_NULL_HANDLE) {
+                XrCompositionLayerFlags         compositionLayerPassthroughFlags = {};
+                XrCompositionLayerPassthroughFB compositionLayerPassthrough      = {XR_TYPE_COMPOSITION_LAYER_PASSTHROUGH_FB};
+                compositionLayerPassthrough.next                                 = nullptr;
+                compositionLayerPassthrough.flags                                = compositionLayerPassthroughFlags;
+                compositionLayerPassthrough.space                                = XR_NULL_HANDLE;
+                compositionLayerPassthrough.layerHandle                          = mPassthroughLayer;
+                layers.push_back(reinterpret_cast<XrCompositionLayerBaseHeader*>(&compositionLayerPassthrough));
```

Comment:
Completely missed that. Yes that's right! It was probably working because all the local variables are allocated with one sub at the function start, but definitely a nasty bug.

Reply:
Thanks for the catch! Fixed.